tasks/request_files/request_files.py

```diff
@@ -235,263 +245,213 @@ def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float,
                     LOGGER.warning(err)
                     a_file[FILE_ERROR_MESSAGE_KEY] = str(err)
 
         attempt = attempt + 1
-        if attempt <= max_retries + 1:  # Only sleep if not on last attempt. # todo: Use backoff code. ORCA-201
+        if (
+            attempt <= max_retries + 1
+        ):  # Only sleep if not on last attempt. # todo: Use backoff code. ORCA-201
             # Check for early completion.
-            if all(a_file[FILE_SUCCESS_KEY] for a_file in granule[GRANULE_RECOVER_FILES_KEY]):
+            if all(
+                a_file[FILE_SUCCESS_KEY]
+                for a_file in granule[GRANULE_RECOVER_FILES_KEY]
+            ):
                 break
             time.sleep(retry_sleep_secs)
 
+    files = []
     any_error = False
     for a_file in granule[GRANULE_RECOVER_FILES_KEY]:
         # if any file failed, the whole granule will fail
         if not a_file[FILE_SUCCESS_KEY]:
             any_error = True
-            # If this is reached, that means there is no entry in the db for file's status.
-            post_status_for_file_to_queue(
-                job_id, granule_id, os.path.basename(a_file[FILE_KEY_KEY]),
-                a_file[FILE_KEY_KEY],
-                a_file[FILE_DEST_BUCKET_KEY],
-                ORCA_STATUS_FAILED,
-                a_file.get(FILE_ERROR_MESSAGE_KEY, None),
-                request_time,
-                datetime.now(timezone.utc).isoformat(),
-                None,
-                RequestMethod.POST,
-                db_queue_url,
-                max_retries,
-                retry_sleep_secs)
+            files["status_id"] = shared_recovery.OrcaStatus.FAILED.value
+            files["error_message"] = a_file[FILE_ERROR_MESSAGE_KEY]
+            files["completion_time"] = datetime.now(timezone.utc).isoformat()
+        
+        files.append(
+            {
+                "filename": os.path.basename(a_file[FILE_KEY_KEY]),
+                "key_path": a_file[FILE_KEY_KEY],
+                "restore_destination": a_file[FILE_DEST_BUCKET_KEY],
+                "status_id": files["status_id"],
+                "error_message": files["error_message"],
+                "request_time": datetime.now(timezone.utc).isoformat(),
+                "last_update": datetime.now(timezone.utc).isoformat(),
+                "completion_time": files["completion_time"]
+            }
+        )
+        # send message to DB SQS
+        # post to DB-queue. Retry using exponential delay if it fails
+        for retry in range(max_retries + 1):
+            try:
+                shared_recovery.create_status_for_job(
+                    job_id, granule_id, glacier_bucket, files, db_queue_url
+                )
+                break
+            except Exception as ex:
+                LOGGER.error(
+                    f"Ran into error posting to SQS {retry+1} time(s) with exception {ex}"
+                )
+                time.sleep(2)
```

Comment:
Ideally we should use the exponential backoff code. @bhazuka, got a good template?